pkg/privateapi/leadership.go

```diff
@@ -194,6 +194,46 @@ func (s *Server) BecomeLeader(ctx context.Context) ([]PeerId, string, error) {
 	return acked, request.View.LeadershipToken, nil
 }
 
+func (s *Server) AssertLeadership(ctx context.Context, leadershipToken string) error {
+	// TODO: Should we send a notification if we ourselves would reject it?
+	snapshot, infos := s.snapshotHealthy()
+
+	request := &LeaderNotificationRequest{}
+
+	request.View = &View{}
+	for _, info := range infos {
+		request.View.Healthy = append(request.View.Healthy, info)
+	}
+	request.View.Leader = &s.myInfo
+	request.View.LeadershipToken = leadershipToken
+
+	for peerID := range snapshot {
+		conn, err := s.GetPeerClient(peerID)
+		if err != nil {
+			return fmt.Errorf("error getting peer client for %s: %w", peerID, err)
+		}
+
+		peerClient := NewClusterServiceClient(conn)
+
+		timeoutCtx, cancel := context.WithTimeout(ctx, 10*time.Second)
+		response, err := peerClient.LeaderNotification(timeoutCtx, request)
+		cancel()
+		if err != nil {
+			return fmt.Errorf("error sending leader assertion to %s: %w", peerID, err)
+		}
+
+		if response.View != nil {
+			s.addPeersFromView(response.View)
+		}
+
+		if !response.Accepted {
+			return fmt.Errorf("our leadership assertion was not accepted by peer %q: %v", peerID, response)
```

Comment:
Here we print the peer ID in quotation marks, but at lines 213 and 222 above we don't. I prefer the quoted form as you used here.

Reply:
Good call - I fixed up this function, I want to keep change this fairly minimal but we can do another PR to address other occurrences in existing code.  Thanks seh@!